### ISSUE TITLE
Disable scrolling when the modal is open

### ISSUE THREAD

**Issue by raphaelbernhart** [dargmuesli/nuxt-cookie-control#229]
### Describe the feature

At least for me on mac the website is still scrollable even if the modal is opened. IMO it is common sense that the website scroll should be blocked when a modal is opened.

FYI - I'm open to create a MR for this but please add some feedback of your thoughts
Sidenote for me: keep iOS in mind when developing this feature.

### Additional information

- [X] Would you be willing to help implement this feature?

### Final checks

- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by dargmuesli** [comment]
Good idea, imho. Consider using [headlessui/dialog](https://headlessui.com/v1/vue/dialog), I'd say :bulb:

**Comment by raphaelbernhart** [comment]
> Good idea, imho. Consider using [headlessui/dialog](https://headlessui.com/v1/vue/dialog), I'd say 💡

Do you think headlessui could bloat the module unnecessarily? IMO installing a ui-library should be discussed as the module over all could then probably need a refactor?

Nevertheless I think the scroll-disable feature can also be added without installing a third-party (just FYI).
What do you think?

**Comment by raphaelbernhart** [comment]
see https://github.com/dargmuesli/nuxt-cookie-control/discussions/228#discussioncomment-10656009

**Comment by kennyhei** [comment]
As a workaround, I did this in my main `.css` file:

```css
/* Disable page scroll when cookie control modal is open */
body:has(.cookieControl__Modal) {
  @apply overflow-hidden;
}
```

**Comment by dargmuesli** [comment]
I've wrapped the modal in a HTML `<dialog>` element:
- https://github.com/dargmuesli/nuxt-cookie-control/pull/304

It's released as version [v10.0.0-beta.3](https://github.com/dargmuesli/nuxt-cookie-control/releases/tag/10.0.0-beta.3).

You can now simply apply the following css to your page and have disabled background scrolling as long as the dialog is open:

```css
body:has(dialog[open]) {
  overflow-y: hidden;
  touch-action: none;
}
```
